### ISSUE TITLE
Checkbox "Don't ask next time" not working

### ISSUE THREAD

**Issue by albertskog** [cavi-au/Consent-O-Matic#488]
When I press "Let us know!" to report a site and in the confirmation dialogue check the box "Don't ask next time", the confirmation still comes up the next time.

It seems to be caused by line 34 in `popup.js`:

```javascript
confirmationInput.addEventListener("input", () => {
    settings.skipSubmitConfirmation = confirmationInput.checked;
    GDPRConfig.setValuesFlags(settings);
});
```

`GDPRConfig.setValuesFlags()` does not exist, I think it should be `GDPRConfig.setDebugValues()`.

I confirmed that making this change fixes the issue I was having on Firefox.

**Comment by jhertel** [comment]
I have had the same problem ever since switching to Firefox (currently version 138, Windows 11 and Linux Mint 21.3) from Chrome (also Windows 11). 

At first I thought the problem was only in Firefox (since it seemed to work well in Chrome and didn't ask for confirmation when I submitted websites), but then I tested it in a new profile in Chrome, and, lo and behold, it doesn't work there either. It keeps asking for confirmation there too, no matter how many times you tell it not to ask again. So it only "worked in Chrome" for me because I must have set it a long time ago using a previous version of Consent-O-Matic.

I checked with these two websites (you need some websites which aren't currently supported by Consent-O-Matic to check it):
https://www.netzleitungen.com/
https://ascentoptics.com/

So I can confirm that in fact the "Don't ask next time" option doesn't work both in Firefox (138.0.4.391, Windows 11 and Linux Mint 21.3) and Chrome (136.0.7103.114, Windows 11). Which is probably unsurprising for those who understand the code; I didn't look into it.

**Comment by jhertel** [comment]
@albertskog Is there an easy (or relatively easy) way to make this work until the maintainers accept your submitted patch? Like do you have a file that can be used to install a patched version?

**Comment by albertskog** [comment]
@jhertel Yes, you can clone [my fork](https://github.com/albertskog/Consent-O-Matic/tree/master), verify my tiny change and follow the instructions under ["Building from Source"](https://github.com/cavi-au/Consent-O-Matic?tab=readme-ov-file#building-from-source) in the readme.

I would send you the file but I don't recommend you install an unsigned extension from a stranger on the Internet :)

**Comment by jhertel** [comment]
@albertskog Oh, I trust you completely! If you take your time to report a bug like this, in the kind and proper way you did it, with even a patch supplied, you are certainly a good and trustworthy human being in my book. 

But thank you for showing me the way to the ocean so I could figure out how to fish. :-) That was quite a ride. Among other things, I had to learn how to install a newer version of node without overriding the old system version, had to install nvm, had to figure out how to install an unsigned extension from a file in Firefox (which takes you for a trip through about:debugging#/runtime/this-firefox), etc. I am amazed at how much time I have spent on getting it to remember that little checkmark. :-) ChatGPT was also quite helpful on my way, even though you have to fact-check what it says before you actually do it.

But now it works for me, at least on Linux! Next step will be to make it work on Windows too. But I hope I can at least use the same zip file that I created on Linux.

Thanks for your help! :-)
